src/prezzemolo/graph.py

```diff
@@ -14,12 +14,35 @@
 
 
 from collections import deque
+from dataclasses import dataclass
+from queue import PriorityQueue
 from typing import Deque, Dict, Generic, Iterator, List, Optional, Set
 
 from prezzemolo.utility import ValueType
 from prezzemolo.vertex import Vertex
 
 
+@dataclass(frozen=True)
+class _ShortestDistance(Generic[ValueType]):
+    distance: float
+    vertex: Vertex[ValueType]
+
+    # __lt__ and __gt__ are needed for PriorityQueue
+    def __lt__(self, other: object) -> bool:
+        if not other:
+            return False
+        if not isinstance(other, _ShortestDistance):
+            raise TypeError(f"Operand is not class _ShortestDistance: {repr(other)}")
+        return self.distance < other.distance
+
+    def __gt__(self, other: object) -> bool:
+        if not other:
+            return False
+        if not isinstance(other, _ShortestDistance):
+            raise TypeError(f"Operand is not class _ShortestDistance: {repr(other)}")
+        return self.distance > other.distance
```

Comment:
A simpler option would be:
```suggestion
        return not self.__lt__(other)
```